source/Calamari.Common/Features/Discovery/AccountAuthenticationDetails.cs

```diff
@@ -17,14 +16,6 @@ public AccountAuthenticationDetails
             AccountDetails = accountDetails;
         }
 
-        public AccountAuthenticationDetails(
-            string accountId, TAccountDetails accountDetails)
-        {
-            Type = null;
-            AccountId = accountId;
-            AccountDetails = accountDetails;
-        }
-        
         public string? Type { get; set; }
```

Comment:
If the `Type` property is always going to be supplied by Server now (and we only have one constructor with a non-nullable parameter), then we might consider making this non-nullable?